### ISSUE TITLE
Allow a trigger before enabling SHLVL (LP_ENABLE_SHLVL)

### ISSUE THREAD

**Issue by mathieumd** [liquidprompt/liquidprompt#821]
### Description
I discovered [LP_ENABLE_SHLVL](https://liquidprompt.readthedocs.io/en/stable/config.html#LP_ENABLE_SHLVL) today, nice!

But, as I'm always running my shells under Tmux, the bold green mark is always present ("└2"), which distract me. And, more importantly, I would not notice when I'm actually running a subshell ("└3")!

Would it be possible to display the mark only when `$SHLVL` is greater than 2, by example?

### How will this be useful?
Warn about sub shells only when the user needs it.

**Comment by Rycieos** [comment]
I suppose a `LP_SHLVL_THRESHOLD` value would make sense. I think that's a valid request.

As a work around, you could add something to your bash/zshrc that decrements `$SHLVL` by one when it detects its parent is Tmux.

**Comment by mathieumd** [comment]
> As a work around, you could add something to your bash/zshrc that decrements `$SHLVL` by one when it detects its parent is Tmux.

Tried this in `.bashrc`, but then `$SHLVL` is always equal to 1, regardless of number of sub shells!

```sh
if [ "x$TMUX" != "x" ]; then
  SHLVL="$(( SHLVL - 1 ))"
fi
```

```
14:39:16 [mathieu:~] $ bash
14:39:18 [mathieu:~] $ echo $SHLVL
1
14:39:22 [mathieu:~] $ bash
14:39:24 [mathieu:~] $ echo $SHLVL
1
14:39:30 [mathieu:~] $ bash
14:39:32 [mathieu:~] $ echo $SHLVL
1
14:39:34 [mathieu:~] $
exit
14:39:37 [mathieu:~] 5s $
exit
14:39:38 [mathieu:~] 14s $
exit
14:39:39 [mathieu:~] 21s $
```

While without the decrement code, it works well:

```
14:40:49 [mathieu:~]└2 $ bash
14:40:52 [mathieu:~]└3 $ bash
14:40:53 [mathieu:~]└4 $ bash
14:40:54 [mathieu:~]└5 $
exit
14:40:54 [mathieu:~]└4 $
exit
14:40:55 [mathieu:~]└3 2s $
exit
14:40:56 [mathieu:~]└2 5s $
```

**Comment by Rycieos** [comment]
> Tried this in `.bashrc`, but then `$SHLVL` is always equal to 1, regardless of number of sub shells!

Weird, it works perfectly for me if I add this to my `.bashrc`:
```sh
if [[ -n ${TMUX-} ]]; then
  SHLVL=$(( SHLVL - 1 ))
fi
```
Of course, if you then run `bash` or another shell inside that Tmux again, since `$TMUX` is still set, the level will still not increment. And if you ever source `.bashrc` it will decrement the level again. The best way I can think to fix that is to run `tmux` as your login shell, or to invoke Tmux with SHLVL set to 0:
```sh
SHLVL=0 tmux
```

**Comment by mathieumd** [comment]
Thanks for your suggestions @Rycieos, but I can't make them to work on my system (Yep, I'm running Bash inside Tmux, good catch!).

Anyway, it's not blocking, and I can wait, and hope `LP_SHLVL_THRESHOLD` will be implemented one day! ;)

**Comment by Rycieos** [comment]
`LP_SHLVL_THRESHOLD` was released as part of [v2.2.0](https://github.com/liquidprompt/liquidprompt/releases/tag/v2.2.0).
